arm-cognitiveservices/2017-04-18/swagger/cognitiveservices.json

```diff
@@ -0,0 +1,1055 @@
+          },
+          {
+            "$ref": "#/parameters/locationParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CognitiveServicesAccountsCreationSettingResult"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CognitiveServicesAccountCreateParameters": {
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "Required. Gets or sets the SKU of the resource."
+        },
+        "kind": {
+          "$ref": "#/definitions/CognitiveServicesAccountKind",
+          "description": "Required. Gets or sets the Kind of the resource."
+        },
+        "location": {
+          "type": "string",
+          "description": "Required. Gets or sets the location of the resource. This will be one of the supported and registered Azure Geo Regions (e.g. West US, East US, Southeast Asia, etc.). The geo region of a resource cannot be changed once it is created, but if an identical geo region is specified on update the request will succeed."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Gets or sets a list of key value pairs that describe the resource. These tags can be used in viewing and grouping this resource (across resource groups). A maximum of 15 tags can be provided for a resource. Each tag must have a key no greater than 128 characters and value no greater than 256 characters."
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CognitiveServicesAccountPropertiesCreateParameters",
+          "description": "Must exist in the request. Must not be null."
+        }
+      },
+      "required": [
+        "sku",
+        "kind",
+        "location",
+        "properties"
+      ],
+      "description": "The parameters to provide for the account."
+    },
+    "CognitiveServicesAccountKind": {
+      "type": "string",
+      "description": "Required. Indicates the type of cognitive service account.",
+      "enum": [
+        "Academic",
+        "Bing.Autosuggest",
+        "Bing.Search",
+        "Bing.Speech",
+        "Bing.SpellCheck",
+        "ComputerVision",
+        "ContentModerator",
+        "CustomSpeech",
+        "Emotion",
+        "Face",
+        "LUIS",
+        "Recommendations",
+        "SpeakerRecognition",
+        "Speech",
+        "SpeechTranslation",
+        "TextAnalytics",
+        "TextTranslation",
+        "WebLM"
+      ],
+      "x-ms-enum": {
+        "name": "Kind",
+        "modelAsString": true
+      }
+    },
+    "CognitiveServicesAccountUpdateParameters": {
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "Required. Gets or sets the SKU of the resource."
```

Comment:
Please take a look at: "M2016: For PATCH operation, every property in the model for the request body MUST be optional and the property MUST NOT provide any default value. The absence of default values applies recursively to complex properties in the model definition." (from [https://github.com/Azure/azure-rest-api-specs/blob/62c96876795e600de8369efe42edd3a70f494017/documentation/swagger-checklist.md](https://github.com/Azure/azure-rest-api-specs/blob/62c96876795e600de8369efe42edd3a70f494017/documentation/swagger-checklist.md))